### ISSUE TITLE
NullPointerException renaming table on multi-node cluster

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate#15521]
### CrateDB version

5.6.1

### CrateDB setup information

NB: 3 nodes cluster - cannot repro on single node

### Problem description

Attempting to rename tables on a multi-node cluster as a user other than `crate` seems to result on a `NullPointerException` every time on 5.6.1

### Steps to Reproduce

```sql
-- logged in as a user other than the crate superuser on a multi-node cluster

create table demo.torename (a text);

alter table demo.torename rename to targetname;
```

### Actual Result

```
NullPointerException[Cannot invoke "io.crate.role.metadata.RolesMetadata.equals(Object)" because "oldRolesMetadata" is null]

Error trace
java.lang.NullPointerException: Cannot invoke "io.crate.role.metadata.RolesMetadata.equals(Object)" because "oldRolesMetadata" is null
	at io.crate.role.RoleManagerDDLModifier.transferTablePrivileges(RoleManagerDDLModifier.java:126)
	at io.crate.role.RoleManagerDDLModifier.onRenameTable(RoleManagerDDLModifier.java:57)
	at io.crate.metadata.cluster.DDLClusterStateService.lambda$onRenameTable$5(DDLClusterStateService.java:77)
	at io.crate.metadata.cluster.DDLClusterStateService.applyOnAllModifiers(DDLClusterStateService.java:92)
	at io.crate.metadata.cluster.DDLClusterStateService.onRenameTable(DDLClusterStateService.java:76)
	at io.crate.metadata.cluster.RenameTableClusterStateExecutor.execute(RenameTableClusterStateExecutor.java:148)
	at io.crate.execution.ddl.tables.TransportRenameTableAction$1.execute(TransportRenameTableAction.java:108)
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45)
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:669)
	at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:281)
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:176)
	at org.elasticsearch.cluster.service.MasterService.lambda$doStart$0(MasterService.java:122)
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:170)
	at org.elasticsearch.cluster.service.BatchedTask.run(BatchedTask.java:70)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231)
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Expected Result

```
ALTER OK, -1 rows affected
```

**Comment by hlcianfagna** [comment]
I may be wrong and this may be about clusters upgraded from 5.x and not about multi-node, need to test.

**Comment by matriv** [comment]
@hlcianfagna I'm pretty sure that's the issue (upgrade from < 5.6) was it during a rolling upgrade, or after the cluster has completed upgrade?

**Comment by hlcianfagna** [comment]
@matriv this was after the cluster completed the upgrade, however I must say that for some reason today I am not able to reproduce this anymore

**Comment by matriv** [comment]
You mean you cannot reproduce it on the same cluster, or you cannot reproduce the scenario?

**Comment by matriv** [comment]
@hlcianfagna Thx for reporting, the issue has been fixed and will be available with `5.6.2` hotfix release.
